Issue 683: calculating height of iframe with borders and/or padding

superelement opened the pull request:
Addresses https://github.com/davidjbradshaw/iframe-resizer/issues/683

Includes border top/bottom, plus padding top/bottom, when calculating heights on iframe with `box-sizing: border-box`.

superelement commented:
It already is using getComputedStyle (line 159), but I think I have put an unnecessary check for 'px' string on the border widths. I guess that's why it looks, at first glance, like a regular style property. I'll take a look at that tomorrow.

davidjbradshaw commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "iframe-resizer",
-  "version": "4.2.2",
```
Can you please leave this as 4.2.2. My deploy script will up the version number automagically

superelement replied:
done

davidjbradshaw commented on CHANGELOG.md:
```diff
@@ -1,5 +1,6 @@
 # Version History
 
+- v4.2.3 [#683](https://github.com/davidjbradshaw/iframe-resizer/issues/683) Include border top/bottom, plus padding top/bottom, when calculating heights on iframe with `box-sizing: border-box;`
```
Don’t forget to add a credit link I g back to your GitHub account :)

superelement replied:
Thanks - wasn't sure how links worked at first, but I get it now

superelement commented:
I've removed the check for 'px' on border widths (as it will always be 'px') and realised I also needed to strip off the 'px' from padding values before parsing them (fresh set of eyes). Also added a test using 'rem' units to check it's working as expected with units other than 'px'.

davidjbradshaw commented on src/iframeResizer.js:
```diff
@@ -168,21 +168,20 @@
       }
-      var top = compStyle.borderTopWidth && compStyle.borderTopWidth.indexOf('px') !== -1 ? parseInt(compStyle.borderTopWidth.replace('px', ''), 10) : 0
-      var bot = compStyle.borderBottomWidth && compStyle.borderBottomWidth.indexOf('px') !== -1 ? parseInt(compStyle.borderBottomWidth.replace('px', ''), 10) : 0
+      var top = compStyle.borderTopWidth ? parseInt(compStyle.borderTopWidth.replace('px', ''), 10) : 0
```
I don’t think you need `replace` here, as `parseInt('10px') === 10`

superelement replied:
Oh, right. I didn't realise the 'px' got stripped out automatically. I've pushed an update.
I want to leave the radix (10) in there though, as mozilla docs recommend _"always specify a radix when using parseInt"_ because of some browsers defaulting to other values (it doesn't specify which browsers though unfortunately) - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt

superelement replied:
Hi David. Was there some reason this hasn't been approved yet? Happy to discuss.

davidjbradshaw commented:
Hi, sorry I picked up a bug and spent a week in bed, catching up on stuff
this week

On Tue, 29 Oct 2019 at 23:01, Jim Doyle <notifications@github.com> wrote:

> *@superelement* commented on this pull request.
> ------------------------------
>
> In src/iframeResizer.js
> <https://github.com/davidjbradshaw/iframe-resizer/pull/764#discussion_r340365797>
> :
>
> >        return top + bot
>      }
>
>      function getBorderEnds(compStyle) {
>        if (compStyle.boxSizing !== 'border-box') {
>          return 0;
>        }
> -      var top = compStyle.borderTopWidth && compStyle.borderTopWidth.indexOf('px') !== -1 ? parseInt(compStyle.borderTopWidth.replace('px', ''), 10) : 0
> -      var bot = compStyle.borderBottomWidth && compStyle.borderBottomWidth.indexOf('px') !== -1 ? parseInt(compStyle.borderBottomWidth.replace('px', ''), 10) : 0
> +      var top = compStyle.borderTopWidth ? parseInt(compStyle.borderTopWidth.replace('px', ''), 10) : 0
>
> Hi David. Was there some reason this hasn't been approved yet? Happy to
> discuss.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/davidjbradshaw/iframe-resizer/pull/764?email_source=notifications&email_token=AAEQOESCYXJWS5DWVDPL5TTQRC6EJA5CNFSM4JAORXBKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCJUTI4I#discussion_r340365797>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAEQOESL7SIGY7G3GYH5ZTLQRC6EJANCNFSM4JAORXBA>
> .
>
-- 
Sent from my iPhone.

davidjbradshaw commented:
Released in V4.2.3, sorry for the delay

superelement commented:
Cool, thanks.
